src/tools/impl/tinytex-info.ts

```diff
@@ -23,7 +23,12 @@ export function hasTinyTex(): boolean {
 export function tinyTexInstallDir(): string | undefined {
   switch (Deno.build.os) {
     case "windows":
-      return expandPath(join(getenv("APPDATA", undefined), "TinyTeX"));
+      let appDir = getenv("APPDATA", undefined);
+      // use ProgramData if APPDATA is not set or contains spaces or non-ASCII characters
+      if (!appDir || !appDir.match(/^[!-~]+$/)) {
```

Comment:
Strictly speaking, the range `!-~` doesn't include the full ASCII table. It excludes spaces (the space is right before `!` in the table: https://en.wikipedia.org/wiki/ASCII#Character_set) and other characters that are very unlikely to be present in a file path.